### DISCUSSION TITLE
Access previous version documentation

### DISCUSSION THREAD

**Discussion by e-gloo** [open-source-labs/Svelvet#299] [General]
Hello, would it be possible to give access to the documentation of the previous versions?
The version 7 is great, but some features (like the dynamic anchor) were super useful and it was working out of the box. Until something similar is available in the v7, it would be nice if we could still use the v6 and have access to the documentation.

**Comment by briangregoryholmes** [comment]
Absolutely! We’ll get the old docs back up tomorrow. Dynamic Anchors will be added to v7 in the next version released this weekend! You’ll be able to mix dynamic and static anchors now. If there’s anything else in particular you’re looking for, let us know and we’ll prioritize it.

**Comment by e-gloo** [comment]
Great! Can't wait to try the dynamic anchor this weekend then!

You shouldn't have said that, I have a lot of features that I'm looking for, be ready!  😋
To be quite honest, I am currently testing VueFlow as well, and there are some useful feature from there that I'd like to have in Svelvet as well.

- I would need to be able to save and restore a canvas, I saw it in your roadmap, and I'm really looking forward to it!

- I think it would be very handy to be able to set up a connection out of the context of a node or an anchor. Therefore, a connection would be with the node|node/anchor from both ends, and a list of connections would be like:
```
const connections: TwoWayConnectionType = [
  [[nodeId1], [nodeId2, anchorId2]],
  [[nodeId3], [nodeId4]],
];
```
However, a 3 level array is not very practical, so I suggest moving from array of arrays to array of object, example:
```
const connections: TwoWayConnectionType = [
  {source: [nodeId1], target: [nodeId2, anchorId2]},
  {source: [nodeId3], target: [nodeId4]},
];
```
And then maybe there would be a function signature like `addConnections(connections: TwoWayConnectionType)`. This way, we would be able to add connection programmatically and not only from the UI. Moreover, I think this way, the save feature would be much easier as you wouldn't need to choose a node or an anchor in which to save the connection. [VueFlow edge ref](https://vueflow.dev/guide/edge.html)

- Another handy features that would help the save/restore feature, is to set a node type in the node object and then be able to attach a component to this type during the display. So during the restore, it would be easier to know which component or `NodeWrapper` to call depending on the node type we get - during a loop for example. [VueFlow custom node ref](https://vueflow.dev/guide/node.html#custom-nodes)

- Last thing (I promise), having a path finding for edges that could avoid nodes. So instead of going behind a node, it would go around it. This is just because I'm willing to make an SQL Entity Relation Diagram, and it's much easier to follow foreign key references when you can follow the edge all the way. But it's clearly not a must have, just cool if it's possible. (Maybe I'll dig into your path drawing and try to implement it myself, who knows 👀).

If you want me to open a new discussion for each topic just tell me, but I think, implementing the 2 way connection and the node type would actually be very useful for the save/restore feature, so maybe it's related after all?

Also, are you planing to put the documentation open source? I found some mistakes and I would be happy to help and fix them, just tell me where to do it!

**Comment by briangregoryholmes** [comment]
These are all great and a few of them should come sooner than others. Help on the documentation would be fantastic. The repository for that will be moved to os-labs eventually, but it's currently on my profile [here](https://github.com/briangregoryholmes/docs). You should be able to make PRs.

**Comment by e-gloo** [comment]
Great! Will look into it.

**Comment by briangregoryholmes** [comment]
Old docs back up here: https://www.svelvet.io/v6/installation/

New version of v7 has dynamic anchors functionality. Set via a `dynamic` prop on a Node or Anchor. I'll update the documentation soon.
